src/Signup.js

```diff
@@ -200,7 +207,7 @@ class Register extends Signup {
         return stage.complete().then(function(request) {
             if (request.auth) {
                 console.log("Stage %s is returning an auth dict", stageName);
-                return self._tryRegister(request.auth);
+                return self._tryRegister(request.auth, request.poll_for_success);
```

Comment:
There's nothing special about this, spec-wise: 3pid UI auth works by the validation being done out of band, at which point you retry the auth the get the HS to check the validation has been done: this just takes advantage of the fact that you can retry the auth safely without knowing if the validation has completed, since it will keep 401ing if it hasn't.